source/common/modules/markdown-editor/parser/pandoc-div-span-parser.ts

```diff
@@ -0,0 +1,122 @@
+/**
+ * @ignore
+ * BEGIN HEADER
+ *
+ * Contains:        Pandoc Span and Div Parser
+ * CVM-Role:        InlineParser, BlockParser
+ * Maintainer:      Hendrik Erz
+ * License:         GNU GPL v3
+ *
+ * Description:     This module provides an inline and a block parser for pandoc
+ *                  bracketed spans and fenced divs
+ *
+ * END HEADER
+ */
+
+import type { InlineParser, BlockParser, BlockContext, Line, DelimiterType } from '@lezer/markdown'
+
+const PandocSpanDelimiter: DelimiterType = {}
+
+const pandocSpanClosingRe = /^\](?<attr>\{[ \w\t\-.#:=;"')(]*\})/d
+
+const pandocDivOpeningRe = /^[ \t]*(?<mark>:{3,})[ \t]+(?:(?<name>[\w\-.]+)|(?:(?<class>[\w\-.]+)[ \t]+)?(?<attr>\{[ \w\t\-.#:=;"')(]*\}))\s*$/d
+
+const pandocDivClosingRe = /^[ \t]*(?<mark>:{3,})\s*$/d
+
+export const pandocSpanParser: InlineParser = {
+  name: 'pandoc-span',
+  before: 'Link',
+  parse: (ctx, next, pos) => {
+    if (next === 91) { // 91 === '['
+      ctx.addDelimiter(PandocSpanDelimiter, pos, pos + 1, true, false)
+
+      // Return -1 so that the default link parser can add delimiters
+      return -1
+    }
+
+    if (next !== 93) { // 93 === ']'
+      return -1
+    }
+
+    // There are no valid attributes, so return
+    const match = pandocSpanClosingRe.exec(ctx.text.slice(pos - ctx.offset))
+    if (!match?.indices?.groups) { return -1 }
```

Comment:
Can you move these ifs across the PR into three lines? Makes them a bit more verbose, but it's always easier to read. I'm also curious why the linter rule didn't catch this that I added due to this …